yarn-project/circuits.js/src/keys/derivation.ts

```diff
@@ -41,11 +41,10 @@ export function computeIvskApp(ivsk: GrumpkinPrivateKey, address: AztecAddress)
 }
 
 export function computeOvskApp(ovsk: GrumpkinPrivateKey, address: AztecAddress) {
+  const ovskAppFr = computeAppSecretKey(ovsk, address, 'ov'); // 'ov' is the key prefix for outgoing viewing key
   // Here we are intentionally converting Fr (output of poseidon) to Fq. This is fine even though a distribution of
   // P = s * G will not be uniform because 2 * (q - r) / q is small.
-  return GrumpkinPrivateKey.fromBuffer(
-    poseidon2Hash([address.toField(), ovsk.high, ovsk.low, GeneratorIndex.OVSK_M]).toBuffer(),
-  );
+  return GrumpkinPrivateKey.fromBuffer(ovskAppFr.toBuffer());
```

Comment:
This was a bug because the order of values in the preimage was different here from the generic `computeAppSecretKey`. This resulted in the outgoing not getting decrypted in the `e2e_2_pxes` (which I modified to check outgoing). This was not caught up until now because the outgoing were not tested in an e2e setting (we could not fetch them before this PR).